erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -570,6 +571,33 @@ def validate_proj_cust(self):
 			if not res:
 				throw(_("Customer {0} does not belong to project {1}").format(self.customer,self.project))
 
+	def set_inter_company_account(self):
+		"""
+			Set intercompany account for inter warehouse transactions
+			This account will be used in case billing company and internal customer's
+			representation company is same
+		"""
+
+		if self.is_internal_transfer() and not self.unrealized_profit_loss_account:
+			unrealized_profit_loss_account = frappe.db.get_value('Company', self.company, 'unrealized_profit_loss_account')
+
+			if not unrealized_profit_loss_account:
+				msg = _("Please select Unrealized Profit / Loss account or add default Unrealized Profit / Loss account account for company {0}").format(
+						frappe.bold(self.company))
+				frappe.throw(msg)
+
+			self.unrealized_profit_loss_account = unrealized_profit_loss_account
+
+	def is_internal_transfer(self):
+		"""
+			It will an internal transfer if its an internal customer and representation
+			company is same as billing company
+		"""
+		if self.is_internal_customer and (self.represents_company == self.company):
+			return True
+
+		return False
+
```

Comment:
This is repetitive, the same as PI. Move those functions to accounts_controller.py?